extensions/amp-analytics/0.1/events.js

```diff
@@ -619,24 +621,31 @@ export class ScrollEventTracker extends EventTracker {
 
   /**
    * Function to handle scroll events from the Scroll manager
-   * @param {!Object<number,boolean>} boundsV
    * @param {!Object<number,boolean>} boundsH
+   * @param {!Object<number,boolean>} boundsV
+   * @param {boolean} useInitialPageSize
    * @param {function(!AnalyticsEvent)} listener
    * @param {!Object} e
    * @private
    */
-  scrollHandler_(boundsV, boundsH, listener, e) {
+  scrollHandler_(boundsH, boundsV, useInitialPageSize, listener, e) {
     // Calculates percentage scrolled by adding screen height/width to
     // top/left and dividing by the total scroll height/width.
     this.triggerScrollEvents_(
       boundsV,
-      ((e.top + e.height) * 100) / e./*OK*/ scrollHeight,
+      ((e.top + e.height) * 100) /
+        (useInitialPageSize
```

Comment:
Can you do a single top-destructure or assignment rather than inlining the prop use?

either:

```js
const {
  scrollWidth,
  scrollHeight,
} = useInitialLength ? e.initialScrollLength : e;
scrollHeight;
scrollWidth;
```

or:

```js
const scrollLength = useInitialLength ? e.initialScrollLength : e;
scrollLength.scrollHeight;
scrollLength.scrollWidth;
```